### ISSUE TITLE
Invalid search filter causes an unwrap panic

### ISSUE THREAD

**Issue by lawliet89** [inejge/ldap3#19]
Consider some incorrect search filter:

```rust
    let (result, _) = connection
        .search("dc=example,dc=com", Scope::Subtree, "", vec!["*"])?
        .success()?;
```

This will cause a panic. I believe it is due to this [line](https://github.com/inejge/ldap3/blob/db0773359fed558c0318c7267654aa6f3c8142b2/src/search.rs#L439).

**Comment by inejge** [comment]
Yes, it's one of the places where error handling is not what it should be. Fixed in 25b99eea70e51d9a994d9c144191d5213da188bc. I won't cut a new release since this should be low-impact, and there are a couple of other issues I'd like to address first. I'll leave the issue open until then.

**Comment by inejge** [comment]
Very belatedly, closing this. Error handling in general has been overhauled, and this case is included.
